src/api/MessageAccessories.tsx

```diff
@@ -16,43 +16,45 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
 
+import ErrorBoundary from "@components/ErrorBoundary";
 import { JSX } from "react";
 
-export type AccessoryCallback = (props: Record<string, any>) => JSX.Element | null | Array<JSX.Element | null>;
-export type Accessory = {
-    callback: AccessoryCallback;
+export type MessageAccessoryFactory = (props: Record<string, any>) => JSX.Element | null | Array<JSX.Element | null>;
+export type MessageAccessory = {
+    render: MessageAccessoryFactory;
     position?: number;
 };
 
-export const accessories = new Map<String, Accessory>();
+export const accessories = new Map<string, MessageAccessory>();
 
-export function addAccessory(
+export function addMessageAccessory(
     identifier: string,
-    callback: AccessoryCallback,
+    render: MessageAccessoryFactory,
     position?: number
 ) {
     accessories.set(identifier, {
-        callback,
+        render,
         position,
     });
 }
 
-export function removeAccessory(identifier: string) {
+export function removeMessageAccessory(identifier: string) {
     accessories.delete(identifier);
 }
 
 export function _modifyAccessories(
     elements: JSX.Element[],
     props: Record<string, any>
 ) {
-    for (const accessory of accessories.values()) {
-        let accessories = accessory.callback(props);
-        if (accessories == null)
-            continue;
+    for (const [key, accessory] of accessories.entries()) {
+        const res = (
+            <ErrorBoundary message={`Failed to render ${key} Message Accessory`} key={key}>
+                <accessory.render {...props} />
+            </ErrorBoundary>
+        );
 
-        if (!Array.isArray(accessories))
-            accessories = [accessories];
-        else if (accessories.length === 0)
+        const accessories = Array.isArray(res) ? res : [res];
+        if (!accessories.length)
```

Comment:
```suggestion
        const accessories = Array.isArray(res) ? res : [res];
        if (!accessories.length)
```

This wont ever be an array because it's wrapped in an ErrorBoundary